### DISCUSSION TITLE
ESP32 Lyrat resets using SD card

### DISCUSSION THREAD

**Discussion by TschebullE** [pschatzmann/arduino-audiokit#58] [Q&A]
Hello, I tried to use an example for the Audiokit (streams-sd-audiokit) with the ESP32 Lyrat 4.3 board. The SW starts but when starting to play the file it resets...

11:35:47.359 -> rst:0x8 (TG1WDT_SYS_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
11:35:47.359 -> configsip: 0, SPIWP:0xee
11:35:47.359 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
11:35:47.359 -> mode:DIO, clock div:1
11:35:47.359 -> load:0x3fff0030,len:1344
11:35:47.359 -> load:0x40078000,len:13864
11:35:47.405 -> load:0x40080400,len:3608
11:35:47.405 -> entry 0x400805f0
11:35:48.032 -> [I] AudioTypes.h : 56 - sample_rate: 44100
11:35:48.032 -> [I] AudioTypes.h : 57 - channels: 2
11:35:48.032 -> [I] AudioTypes.h : 58 - bits_per_sample: 16
11:35:48.032 -> [I] I2SConfig.h : 104 - rx/tx mode: TX_MODE
11:35:48.032 -> [I] I2SConfig.h : 105 - port_no: 0
11:35:48.032 -> [I] I2SConfig.h : 106 - is_master: Master
11:35:48.032 -> [I] I2SConfig.h : 107 - sample rate: 44100
11:35:48.032 -> [I] I2SConfig.h : 108 - bits per sample: 16
11:35:48.032 -> [I] I2SConfig.h : 109 - number of channels: 2
11:35:48.065 -> [I] I2SConfig.h : 110 - i2s_format: I2S_STD_FORMAT
11:35:48.065 -> [I] I2SConfig.h : 112 - auto_clear:0
11:35:48.065 -> [I] I2SConfig.h : 119 - buffer_count:10
11:35:48.065 -> [I] I2SConfig.h : 120 - buffer_size:512
11:35:48.065 -> [I] I2SConfig.h : 124 - pin_bck: 14
11:35:48.065 -> [I] I2SConfig.h : 125 - pin_ws: 15
11:35:48.065 -> [I] I2SConfig.h : 126 - pin_data: -1515870811
11:35:48.065 -> [W] AudioKit.h : 54 - sd_active = true
11:35:48.099 -> [I] AudioKit.h : 203 - Channels 2->2
11:35:48.099 -> [I] AudioKit.h : 600 - void audio_tools::AudioKitStream::setupActions()
11:35:48.099 -> [I] AudioKit.h : 350 - void audio_tools::AudioKitStream::addAction(int, void (*)(bool, int, void*), void*)
11:35:48.099 -> [I] AudioActions.h : 42 - ActionLogic::add pin: 39 / logic: 0
11:35:48.099 -> [I] AudioActions.h : 52 - pin 39 -> INPUT_PULLUP
11:35:48.099 -> [I] AudioActions.h : 42 - ActionLogic::add pin: 19 / logic: 2
11:35:48.145 -> [I] AudioActions.h : 55 - pin 19 -> INPUT
11:35:48.145 -> [W] AudioKit.h : 630 - Volume Buttons ignored because of conflict: 13 
11:35:48.178 -> [I] AudioCopy.h : 69 - buffer_size=1024
11:35:48.178 -> [I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops

and after about 1 Second it resets....

I have tried differnet examples but I did not get any with SD card running.

**Comment by pschatzmann** [comment]
Are you sure that you got the pin settings right as documented ?
If you have a sketch with a kit.processActions(); in the loop did you comment this out to prevent any pin clonflicts ?

**Comment by TschebullE** [comment]
if, i use the exanple "player-sd-audiokit" without any changes (i only changed in configfile to 1 (Lyrat Board) it seems that the SD card could be read but the ESP resets always:

6:56:56.707 -> [I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
16:56:57.300 -> ets Jul 29 2019 12:21:46
16:56:57.300 -> 
16:56:57.300 -> rst:0x8 (TG1WDT_SYS_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
16:56:57.347 -> configsip: 0, SPIWP:0xee
16:56:57.347 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
16:56:57.347 -> mode:DIO, clock div:1
16:56:57.347 -> load:0x3fff0030,len:1344
16:56:57.347 -> load:0x40078000,len:13864
16:56:57.347 -> load:0x40080400,len:3608
16:56:57.347 -> entry 0x400805f0
16:56:58.008 -> [I] AudioTypes.h : 56 - sample_rate: 44100
16:56:58.008 -> [I] AudioTypes.h : 57 - channels: 2
16:56:58.008 -> [I] AudioTypes.h : 58 - bits_per_sample: 16
16:56:58.008 -> [I] I2SConfig.h : 104 - rx/tx mode: TX_MODE
16:56:58.008 -> [I] I2SConfig.h : 105 - port_no: 0
16:56:58.008 -> [I] I2SConfig.h : 106 - is_master: Master
16:56:58.008 -> [I] I2SConfig.h : 107 - sample rate: 44100
16:56:58.008 -> [I] I2SConfig.h : 108 - bits per sample: 16
16:56:58.008 -> [I] I2SConfig.h : 109 - number of channels: 2
16:56:58.024 -> [I] I2SConfig.h : 110 - i2s_format: I2S_STD_FORMAT
16:56:58.024 -> [I] I2SConfig.h : 112 - auto_clear:0
16:56:58.024 -> [I] I2SConfig.h : 119 - buffer_count:10
16:56:58.024 -> [I] I2SConfig.h : 120 - buffer_size:512
16:56:58.024 -> [I] I2SConfig.h : 124 - pin_bck: 14
16:56:58.024 -> [I] I2SConfig.h : 125 - pin_ws: 15
16:56:58.024 -> [I] I2SConfig.h : 126 - pin_data: -1515870811
16:56:58.024 -> [W] AudioKit.h : 54 - sd_active = true
16:56:58.056 -> [I] AudioKit.h : 203 - Channels 2->2
16:56:58.056 -> [I] AudioKit.h : 600 - void audio_tools::AudioKitStream::setupActions()
16:56:58.056 -> [I] AudioKit.h : 350 - void audio_tools::AudioKitStream::addAction(int, void (*)(bool, int, void*), void*)
16:56:58.056 -> [I] AudioActions.h : 42 - ActionLogic::add pin: 39 / logic: 0
16:56:58.056 -> [I] AudioActions.h : 52 - pin 39 -> INPUT_PULLUP
16:56:58.056 -> [I] AudioActions.h : 42 - ActionLogic::add pin: 19 / logic: 2
16:56:58.103 -> [I] AudioActions.h : 55 - pin 19 -> INPUT
16:56:58.103 -> [W] AudioKit.h : 630 - Volume Buttons ignored because of conflict: 13 
16:56:58.103 -> [I] AudioPlayer.h : 291 - setVolume(0.700000)
16:56:58.103 -> [I] VolumeStream.h : 183 - setVolume: 0.700000
16:56:58.103 -> [I] VolumeStream.h : 183 - setVolume: 0.700000
16:56:58.135 -> [I] MetaDataID3.h : 556 - virtual void audio_tools::MetaDataID3::begin()
16:56:58.135 -> [I] AudioSourceSD.h : 64 - selectStream: 0
16:56:58.167 -> [I] AudioSourceSD.h : 68 - Using file //02/001Natural_River.mp3
16:56:58.167 -> [I] AudioCopy.h : 69 - buffer_size=1024
16:56:58.202 -> [I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
16:56:58.202 -> [I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops

I do not know who to prevent the conflict on PIN 13 because it is used for touch and SD card function as shown in the schematics

**Comment by pschatzmann** [comment]
If you look at the log, pin 13 is not used by the buttons, so we are fine and this is not the issue.
It does not help that you don't get any stacktrace to analyze.
Another guess, maybe something is wrong with your mp3 file. Did you try different files ? or file types (e.g. wav)
I am surprised that the file name starts with //
